docs/DeveloperGuide.md

```diff
@@ -1,35 +1,306 @@
 # Developer Guide
 
+## Table of Contents
+
+* Table of Contents
+{:toc}
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Adapted from AddressBook Level 3 Developer Guide (https://se-education.org/addressbook-level3/DeveloperGuide.html)
+
+## Setup
+
+### Setting up the project on your computer
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use **JDK 11**.
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.<br>
+   - Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**: Run `Main` and try a few commands.
+
+## Design
+
+> The `.puml` files used to create the diagrams in this guide can be found in the [diagrams](diagrams) folder.
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
```

Comment:
I agree with the reviewers above, the architecture diagram is neatly done and very understandable! Nice!